src/middlewared/middlewared/plugins/failover_/event.py

```diff
@@ -127,14 +156,15 @@ def event(self, ifname, event):
 
         refresh = True
         try:
-            return self._event(ifname, event)
+            job = self._event(ifname, event)
+            return job
         except IgnoreFailoverEvent:
             refresh = False
         finally:
             # refreshing the failover status can cause delays in failover
             # there is no reason to refresh it if the event has been ignored
             if refresh:
-                self.run_call('failover.status_refresh')
+                self.middleware.create_task(self.refresh_failover_status(job.id, event))
```

Comment:
If another exception happens (not IgnoreFailoverEvent) then `refresh` will be `True` but `job` won't be defined